integration_tests/custom_matcher_stack_trace/__tests__/custom_matcher.test.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+'use strict';
+
+function toCustomMatch(callback, expectation) {
+  const actual = callback();
+
+  if (actual !== expectation) {
+    return {
+      message: () => `Expected "${expectation}" but got "${actual}"`,
+      pass: false,
+    };
+  } else {
+    return {pass: true};
+  }
+}
+
+expect.extend({
+  toCustomMatch,
+});
+
+describe('Custom matcher', () => {
+  it('passes', () => {
+    // This expectation should pass
+    expect(() => 'foo').toCustomMatch('foo');
+  });
+
+  it('fails', () => {
+    expect(() => {
+      // This expectation should fail,
+      // Which is why it's wrapped in a .toThrow() block.
+      expect(() => 'foo').toCustomMatch('bar');
+    }).toThrow();
+  });
+
+  it('preserves error stack', () => {
+    const foo = () => bar();
+    const bar = () => baz();
+    const baz = () => {
+      throw Error('qux');
```

Comment:
Should this be `throw new Error`? I'm not sure if there's a semantical difference...

Reply:
No difference that I'm aware of.

> ### 15.11.1 The Error Constructor Called as a Function # Ⓣ 
> When `Error` is called as a function rather than as a constructor, it creates and initialises a new `Error` object. Thus the function call `Error(…)` is equivalent to the object creation expression `new Error(…)` with the same arguments.

source: http://ecma-international.org/ecma-262/5.1/#sec-15.11.1